diff --git a/meson.build b/meson.build
index faf69f7b77..df74564328 100644
--- a/meson.build
+++ b/meson.build
@@ -1360,9 +1360,15 @@ if features['vulkan'] or (features['gl'] and features['egl'])
     sources += files('video/filter/vf_gpu.c')
 endif
 
-if features['d3d11'] or features['egl-angle-win32'] or (features['vulkan'] and win32)
+d3d_hwaccel = get_option('d3d-hwaccel').require(
+    win32,
+    error_message: 'the os is not win32!',
+)
+features += {'d3d-hwaccel': d3d_hwaccel.allowed()}
+features += {'dxgi-debug-d3d11':
+    win32 and cc.has_header_symbol('d3d11sdklayers.h', 'DXGI_DEBUG_D3D11')}
+if features['d3d-hwaccel'] or features['d3d11'] or features['egl-angle-win32'] or (features['vulkan'] and win32)
     sources += files('video/out/gpu/d3d11_helpers.c')
-    features += {'dxgi-debug-d3d11': cc.has_header_symbol('d3d11sdklayers.h', 'DXGI_DEBUG_D3D11')}
 endif
 
 # hwaccel
@@ -1403,11 +1409,6 @@ if features['android-media-ndk']
     sources += files('video/out/hwdec/hwdec_aimagereader.c')
 endif
 
-d3d_hwaccel = get_option('d3d-hwaccel').require(
-    win32,
-    error_message: 'the os is not win32!',
-)
-features += {'d3d-hwaccel': d3d_hwaccel.allowed()}
 if features['d3d-hwaccel']
     sources += files('video/d3d.c',
                      'video/filter/vf_d3d11vpp.c')
